### PR TITLE
Support for FritzBox 7412

### PR DESCRIPTION
**Pull request by AndyX90**
Initial Support for AVM - FritzBox 7412
This is the work of Valentin Spreckels as of
https://patchwork.ozlabs.org/patch/883955/
and
https://patchwork.ozlabs.org/patch/883954/

### CODE REVIEW

File: target/linux/lantiq/files-4.14/arch/mips/boot/dts/FRITZ7412.dts [hunk 1]
```diff
@@ -0,0 +1,241 @@
+/dts-v1/;
+
+#include "vr9.dtsi"
+
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/mips/lantiq_rcu_gphy.h>
+
+/ {
+	compatible = "avm,fritz7412", "lantiq,xway", "lantiq,vr9";
+	model = "1&1 DSL-Modem / 1&1 WLAN-Router";
```
Comment (blocktrron) [inline target/linux/lantiq/files-4.14/arch/mips/boot/dts/FRITZ7412.dts]: Name of the Device should be "AVM FRITZ!Box 7412".

The Device might be exclusively available thru 1&1 but the backside of the device reads the AVM naming.

(I know this is very inconsistent in the lantiq target)

File: target/linux/lantiq/image/Makefile [hunk 2]
```diff
@@ -664,6 +664,18 @@ define Device/avm_fritz7360sl
...
 TARGET_DEVICES += avm_fritz7360sl
 
+define Device/avm_fritz7412
+  $(Device/AVM)
+  $(Device/NAND)
+  BOARD_NAME := FRITZ7412
+  DEVICE_DTS := FRITZ7412
+  KERNEL_SIZE := 4096k
+  IMAGE_SIZE := 49152k
+  DEVICE_TITLE := 1&1 DSL-Modem / WLAN-Router - FRITZ7412
```
Comment (blocktrron) [inline target/linux/lantiq/image/Makefile]: Same here - should be "AVM FRITZ!Box 7412"

File: target/linux/lantiq/files-4.14/arch/mips/boot/dts/FRITZ7412.dts [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+		compatible = "lantiq,xrx200-mdio";
+		phy11: ethernet-phy@11 {
+			reg = <0x11>;
+			compatible = "lantiq,phy11g", "ethernet-phy-ieee802.3-c22";
+		};
+	};
+};
+
+&vmmc {
+	status = "okay";
```
Comment (blocktrron) [inline target/linux/lantiq/files-4.14/arch/mips/boot/dts/FRITZ7412.dts]: I'm not entirely familiar with the lantiq target - but gpio property seems to be missing for telephony ports to work.

File: target/linux/lantiq/image/Makefile (hunk 2 above)
Comment (AndyX90) [reply target/linux/lantiq/image/Makefile]: Thanks, you are right! I changed both to "AVM FRITZ!Box 7412".

**Comment by blocktrron** [comment]
You should also squash your commits and follow the guidelines from https://openwrt.org/submitting-patches

**Pull request closed without merging**